File: archlucid-ui/src/lib/digest/exec-digest-sponsor-deep-link-operator-links.ts
/** Mirrors `ExecDigestSponsorDeepLinkOperatorLinks` for UI tests (TB-2196). */
export const ExecDigestSponsorDeepLinkOperatorLinks = {
  buildDashboardUrl(operatorBaseUrl: string | null | undefined, token: string): string {
    const relativePath = `/digest/sponsor?token=${encodeURIComponent(token)}`;
    if (!operatorBaseUrl?.trim())
      return relativePath;

    return `${operatorBaseUrl.trim().replace(/\/$/, "")}${relativePath}`;
  },

  buildRunCollateralUrl(
    operatorBaseUrl: string | null | undefined,
    runIdHex: string,
    token: string,
  ): string {
    const normalizedRunIdHex = runIdHex.trim().replace(/-/g, "");
    const relativePath =
      `/digest/sponsor/run/${encodeURIComponent(normalizedRunIdHex)}?token=${encodeURIComponent(token)}`;

    if (!operatorBaseUrl?.trim())
      return relativePath;

    return `${operatorBaseUrl.trim().replace(/\/$/, "")}${relativePath}`;
  },
} as const;

File: ArchLucid.Application/Notifications/Email/ExecDigestSponsorDeepLinkOperatorLinks.cs
namespace ArchLucid.Application.Notifications.Email;

/// <summary>Anonymous operator-shell deep links for weekly digest sponsor CTAs (TB-2196).</summary>
public static class ExecDigestSponsorDeepLinkOperatorLinks
{
    public static string BuildDashboardUrl(string? operatorBaseUrl, string token)
    {
        string relativePath = $"/digest/sponsor?token={Uri.EscapeDataString(token)}";

        if (string.IsNullOrWhiteSpace(operatorBaseUrl))
            return relativePath;

        return $"{operatorBaseUrl.TrimEnd('/')}{relativePath}";
    }

    public static string BuildRunCollateralUrl(string? operatorBaseUrl, string runIdHex, string token)
    {
        string normalizedRunIdHex = runIdHex.Trim().Replace("-", string.Empty, StringComparison.Ordinal);
        string relativePath =
            $"/digest/sponsor/run/{Uri.EscapeDataString(normalizedRunIdHex)}?token={Uri.EscapeDataString(token)}";

        if (string.IsNullOrWhiteSpace(operatorBaseUrl))
            return relativePath;

        return $"{operatorBaseUrl.TrimEnd('/')}{relativePath}";
    }
}
